source/guides/basics/mechanisms.md

```diff
@@ -1,49 +1,150 @@
+
+```dscript_green
+defintion_adjust:
+    type: task
+    script:
+    - if <player.item_in_hand.material.name> != air:
+        - define player_item <player.item_in_hand>
+        - take <player.item_in_hand>
+        - adjust def:player_item "display_name:Adjusted by definition"
+        - give <[player_item]>
+```
+
+This script, when run with an attached player, will take the item they are holding and return it with a modified display name.
 
 ### Properties
 
 "Properties" are a system to track the details of an object that doesn't have a unique identity of its own.
 
 #### Item Properties
 
-The most common place that properties are seen in use is with items.
+The most common place that properties are seen in use is with items. In Denizen, an item is a `material` and a series of mechanisms that compose its properties. For example, if you hit an entity with a freshly-crafted diamond sword in survival mode and then run `/ex narrate <player.item_in_hand>` to view the Denizen construction of that item, you'll see it return `i@diamond_sword[durability=1]`. In this case, `durability` is the only property of the item, and its value is `1`, the amount of durability the item has lost thus far. Multiple properties are separated by semicolons. For example, `i@diamond_sword[durability=1;display_name=Skullbuster]`. <span class="parens">(Remember, [never use raw object notation](/guides/troubleshooting/common-mistakes#don-t-type-raw-object-notation) - these examples are for demonstrative purposes only!)</span>
+
+Items are, fundamentally, their material and their properties. Any item a player has in their inventory can be parsed in this manner, and an item's properties can be read and modified as desired.
 
 #### Material Properties
 
-#### Entities Have Properties Too
+Materials, like items, lack unique identities. A stone block is just a material at a location. A wheat block is only slightly more complicated - it's a material at a location with an `age` property. As you saw above in the `adjustblock` example, using the `age` mechanism allows you to modify that property for a block in the world. When you read block materials from locations, they will return any associated properties. This could be used, for example, to check whether a campfire is sending up smoke signals, or various redstone properties, or even the faces of a mushroom block.
+
+Materials with properties, when parsed by a tag like `<player.cursor_on.material>`, will return an output like `m@wheat[age=7]` for a fully-grown wheat block. In this case, the `age` of the wheat is `7`.
+
+#### Entity Properties
+
+Entities have properties too! Entity properties can be read and modified just like material properties, and you can see a list of all of an entity's properties with the `<EntityTag.describe>` tag. For example, here's a sample output of `/ex narrate <player.describe>` as run by the author of this page:
```

Comment:
Player properties are not properly valid or supported, despite technically existing. A mob or armor stand or something should be used for this instead. (Valid describe output should be able to be stuck into the `spawn` command and work).